kasa/smart/smartdevice.py

```diff
@@ -333,17 +336,18 @@ async def _initialize_modules(self):
                 skip_parent_only_modules and mod in NON_HUB_PARENT_ONLY_MODULES
             ) or mod.__name__ in child_modules_to_skip:
                 continue
-            if (
-                mod.REQUIRED_COMPONENT in self._components
-                or self.sys_info.get(mod.REQUIRED_KEY_ON_PARENT) is not None
+            required_component = cast(str, mod.REQUIRED_COMPONENT)
```

Comment:
`REQUIRED_COMPONENT` is typed with `str | None`, but it should not be none here.